### PR TITLE
NonEmpty ToSchema instance

### PR DESCRIPTION
**Pull request by parsonsmatt**
This would be nice to have :) @lucasdicioccio put this in the #93 issue. It seems good to me so I'm opening a PR for it and will be willing to make further changes.

### CODE REVIEW

**Comment by fizruk** [comment]
Apparently your branch is out-of-date. Apart from that, I'm happy with the patch!

@phadej you have any comments?

**Comment by parsonsmatt** [comment]
The doctests are failing locally for some reason, but aren't displaying any output :man_shrugging: 

I had to vendor an `Arbitrary (NonEmpty a)` instance because `quickcheck` removed it for 2.10.

**Comment by phadej** [comment]
Use `quickcheck-instances`

File: swagger2.cabal [hunk 1]
```diff
@@ -96,6 +98,9 @@ test-suite spec
                   , unordered-containers
                   , vector
                   , lens
+  if !impl(ghc >= 8.0)
+     build-depends:    semigroups >= 0.18.3 && <0.19
```
Comment (phadej) [inline swagger2.cabal]: no need to specify bounds in test-suite (as they are inherited from library). This helps if/when `semigroups-0.19` is released.

File: test/Data/Swagger/Schema/ValidationSpec.hs [hunk 2]
```diff
@@ -260,3 +262,13 @@ instance Arbitrary ZonedTime where
 
 instance Arbitrary UTCTime where
   arbitrary = UTCTime <$> arbitrary <*> fmap fromInteger (choose (0, 86400))
+
+#if MIN_VERSION_QuickCheck(2,10,0)
```
Comment (phadej) [inline test/Data/Swagger/Schema/ValidationSpec.hs]: http://hackage.haskell.org/package/quickcheck-instances

**Comment by parsonsmatt** [comment]
Requested changes done :) Thanks!

**Comment by fizruk** [comment]
@phadej are you fine with changes? I can rebase and merge.

File: test/Data/Swagger/Schema/ValidationSpec.hs [hunk 3]
```diff
@@ -228,35 +231,5 @@ invalidButtonImagesToJSON = genericToJSON defaultOptions
...
-instance (Eq k, Hashable k, Arbitrary k, Arbitrary v) => Arbitrary (HashMap k v) where
-  arbitrary = HashMap.fromList <$> arbitrary
-
-instance (Eq a, Hashable a, Arbitrary a) => Arbitrary (HashSet a) where
-  arbitrary = HashSet.fromList <$> arbitrary
-
-instance Arbitrary T.Text where
-  arbitrary = T.pack <$> arbitrary
-
-instance Arbitrary TL.Text where
```
Comment (phadej) [inline test/Data/Swagger/Schema/ValidationSpec.hs]: all of this gone <3

**Comment by phadej** [comment]
@fizruk let's btw wait until we get dependencies to compile with GHC-8.4.1 and make the release then. Do you have some pending changes, or will master be ok to release?

**Comment by fizruk** [comment]
@phadej there's one more PR (#145), but it's cosmetic.
I'll rebase and merge this PR then and then we can wait for a green light for GHC 8.4.1.

**Comment by phadej** [comment]
@fizruk good plan

**Comment by fizruk** [comment]
Rebased and merged manually in 22e975e.

**Comment by parsonsmatt** [comment]
Thanks! :smile:

**Pull request closed without merging**